ot/da.py

```diff
@@ -960,18 +1092,19 @@ def fit(self, Xs=None, ys=None, Xt=None, yt=None):
         """
 
         # pairwise distance
-        Cost = pairwise_distances(Xs, Xt, metric=self.metric)
+        Cost = dist(Xs, Xt, metric=self.metric)
 
         if self.mode == "semisupervised":
             print("TODO: modify cost matrix accordingly")
             pass
 
         # distribution estimation
-        if self.distribution == "uniform":
-            mu_s = np.ones(Xs.shape[0]) / float(Xs.shape[0])
-            mu_t = np.ones(Xt.shape[0]) / float(Xt.shape[0])
-        else:
-            print("TODO: implement kernelized approach")
+        mu_s = self.distribution_estimation(Xs)
+        mu_t = self.distribution_estimation(Xt)
+
+        # store arrays of samples
+        self.Xs = Xs
+        self.Xt = Xt
```

Comment:
Remove sinkhorn from the BAseEstimator and put it in the SinkhornTransport please.

Keep all teh rest so that we can do super as still have  all the storing and density estimation.